File: src/Plag.Backend.Abstraction/Models/PlagiarismSet.cs
using System;
using System.Text.Json.Serialization;

namespace Xylab.PlagiarismDetect.Backend.Models
{
    public class PlagiarismSet
    {
        [JsonPropertyName("setid")]
        public string Id { get; set; }

        [JsonPropertyName("create_time")]
        public DateTimeOffset CreateTime { get; set; }

        [JsonPropertyName("creator")]
        public int? UserId { get; set; }

        [JsonPropertyName("related")]
        public int? ContestId { get; set; }

        [JsonPropertyName("formal_name")]
        public string Name { get; set; }

        [JsonPropertyName("report_count")]
        public long ReportCount { get; set; }

        [JsonPropertyName("report_pending")]
        public long ReportPending { get; set; }

        [JsonPropertyName("submission_count")]
        public int SubmissionCount { get; set; }

        [JsonPropertyName("submission_failed")]
        public int SubmissionFailed { get; set; }

        [JsonPropertyName("submission_succeeded")]
        public int SubmissionSucceeded { get; set; }
    }
}

File: src/React.UI/src/Models/PlagiarismSet.ts
/**
 * Plagiarism Set Entity
 */
export interface PlagiarismSet {

  /**
   * Plagiarism Set Id
   */
  setid: string;

  /**
   * Create time
   */
  create_time: string;

  /**
   * Creator user Id
   */
  creator: number | null | undefined;

  /**
   * Contest Id
   */
  related: number | null | undefined;

  /**
   * Formal name
   */
  formal_name: string;

  /**
   * The count of total reports
   */
  report_count: number;

  /**
   * The count of pending reports
   */
  report_pending: number;

  /**
   * The count of total submissions
   */
  submission_count: number;

  /**
   * The count of compilation failed submissions
   */
  submission_failed: number;

  /**
   * The count of compilation succeeded submissions
   */
  submission_succeeded: number;
}
